### PR TITLE
Recompose

### PR DESCRIPTION
**Pull request by josepplloo**
## Summary

 - Recompose library implemented.

## Screenshots

recompose

## Trello Card

https://trello.com/c/iaR1lg1R/16-redux-iv-redux-recompose

### CODE REVIEW

File: react/tic-tac-toe/src/redux/login/actions.js [hunk 1]
```diff
@@ -1,24 +1,30 @@
...
-    }
-  },
+  login: values => ({
+    type: actions.LOGIN,
+    target: 'login',
+    service: AuthService.singIn,
+    payload: values,
+    injections: [
+      withPostSuccess((dispatch, response) => {
+        dispatch({
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/redux/login/actions.js]: Here you can do something like ---> ```dispatch(actionsCreator.setauth(response.data))```

File: react/tic-tac-toe/src/redux/login/actions.js [hunk 2]
```diff
@@ -1,24 +1,30 @@
...
+          payload: response.data
+        });
+        saveState({ islogged: response.data });
+      }),
+      withPostFailure((dispatch, response) => {
+        dispatch({ type: actions.LOGIN_FAILURE, payload: response });
+      })
+    ]
+  }),
   setauth: token => ({ type: actions.LOGIN_SUCCESS, payload: token }),
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/redux/login/actions.js]: Change this name to camelCase

File: react/tic-tac-toe/src/redux/login/actions.js [hunk 3]
```diff
@@ -1,24 +1,30 @@
...
+    injections: [
+      withPostSuccess((dispatch, response) => {
+        dispatch({
+          type: actions.LOGIN_SUCCESS,
+          payload: response.data
+        });
+        saveState({ islogged: response.data });
+      }),
+      withPostFailure((dispatch, response) => {
+        dispatch({ type: actions.LOGIN_FAILURE, payload: response });
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/redux/login/actions.js]: I don't know why do you send payload when login failed?, to me if you don't show a message error from backend you should be set a state that show error property on true.

File: react/tic-tac-toe/src/redux/login/reducer.js [hunk 4]
```diff
@@ -1,30 +1,16 @@
-import { reducer as formReducer } from 'redux-form';
-import { combineReducers } from 'redux';
+import { createReducer } from 'redux-recompose';
 
 import { actions } from './actions';
 
 const initialState = {
-  islogged: null,
+  isLoggedIn: null,
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/redux/login/reducer.js]: What is the difference between isLoggedIn and is Auth??

File: react/tic-tac-toe/src/redux/login/reducer.js [hunk 5]
```diff
@@ -1,30 +1,16 @@
...
-    }
-    default:
-      return state;
-  }
-}
-
-const reducers = {
-  form: formReducer,
-  login: loginReducer
+const reducerDescription = {
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/redux/login/reducer.js]: Here try to use primaryActions and override properties show this 👀 https://github.com/Wolox/redux-recompose/tree/master/src/completers/completeReducer

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

File: react/tic-tac-toe/src/redux/login/reducer.js (hunk 4 above)
Comment (josepplloo) [reply react/tic-tac-toe/src/redux/login/reducer.js]: At the moment I have this:
```
const initialState = {
  isLogged: false, \\ for the error message of wrong credentials
  isAuth: false \\ for authorize the user and show the others views, in this case the credentials are correct
};
```

File: react/tic-tac-toe/src/redux/login/reducer.js (hunk 5 above)
Comment (josepplloo) [reply react/tic-tac-toe/src/redux/login/reducer.js]: For sure a great change

File: react/tic-tac-toe/src/app/components/AppRouter/components/AuthInfo/index.js [hunk 6]
```diff
@@ -1,47 +1,48 @@
...
-            {PODIUM_LINK}
-          </Link>
-        </li>
-        <button type="button" onClick={onClick} className={styles.button}>
-          {SINGOUT_MSJ}
-        </button>
-      </ul>
-    ) : (
+function AuthInfo({ isAuth, onError, onClick }) {
+  if (onError) {
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/app/components/AppRouter/components/AuthInfo/index.js]: change it for a if() else if ()

File: react/tic-tac-toe/src/app/components/AppRouter/components/AuthInfo/styles.module.scss [hunk 7]
```diff
@@ -11,13 +11,13 @@
...
   background-color: $black;
   color: $white;
   display: flex;
-  max-width: 80px;
   margin: 0 5px;
-  overflow:auto;
+  max-width: 80px;
+  overflow: auto;
   padding: 10px;
   width: 100%;
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/app/components/AppRouter/components/AuthInfo/styles.module.scss]: Delete it

File: react/tic-tac-toe/src/app/components/AppRouter/components/AuthRoute/index.js [hunk 8]
```diff
@@ -14,24 +14,18 @@ class AuthRoute extends Component {
...
 
   render() {
-    const { islogged, isAuth } = this.props;
-    return islogged && isAuth ? (
-      <Redirect from="/login" to="/game" />
-    ) : (
-      <Login onSubmit={this.handleSubmit} />
-    );
+    const { islogged } = this.props;
+    return islogged ? <Redirect from="/login" to="/game" /> : <Login onSubmit={this.handleSubmit} />;
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/app/components/AppRouter/components/AuthRoute/index.js]: Remember that you have constants for Routes

File: react/tic-tac-toe/src/app/components/AppRouter/index.js [hunk 9]
```diff
@@ -53,21 +55,21 @@ class AppRouter extends Component {
 
 AppRouter.propTypes = {
   isAuth: bool,
-  islogged: bool,
   login: func,
   logout: func,
-  setauth: func
+  setAuth: func,
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/app/components/AppRouter/index.js]: Someone of these props are required??

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

File: react/tic-tac-toe/src/app/screens/Game/styles.module.scss [hunk 10]
```diff
@@ -1,6 +1,17 @@
...
+
 .game {
   display: flex;
   justify-content: center;
+  
+  &-board {
+    border-image: linear-gradient(to bottom, $black, transparent) 1 100%;
+    border-style: solid;
+    border-width: 3px;
+    color: white;
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/app/screens/Game/styles.module.scss]: Use a variable here

File: react/tic-tac-toe/src/app/screens/Podium/index.js [hunk 11]
```diff
@@ -1,14 +1,52 @@
...
 import styles from './styles.module.scss';
 
-function Podium() {
-  return (
-    <div className={styles.container}>
-      <Matches />
-    </div>
-  );
+class Podium extends Component {
+  state = {
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/app/screens/Podium/index.js]: In next pr you will change this for store.

File: react/tic-tac-toe/src/services/MatchesService.js [hunk 12]
```diff
@@ -1,9 +1,12 @@
...
-const getMatches = () =>
-  api
-    .get('/matches')
-    .then(response => response.data)
-    .catch(error => error);
+const MatchesServices = {
+  getMatches: () =>
+    api
+      .get('/matches')
+      .then(response => response.data)
```
Comment (CrisVega08) [inline react/tic-tac-toe/src/services/MatchesService.js]: This change on next pr.

**Review by CrisVega08 — APPROVED** [review]

File: react/tic-tac-toe/src/app/screens/Podium/index.js (hunk 11 above)
Comment (josepplloo) [reply react/tic-tac-toe/src/app/screens/Podium/index.js]: Yes Sir

**Pull request merged**